ocaml/lambda/translcore.ml

```diff
@@ -1124,77 +1126,126 @@ and transl_apply ~scopes
           ap_probe=None;
         }
   in
-  let rec build_apply lam args loc pos ap_mode = function
-    | Omitted { mode_closure; mode_arg; mode_ret; sort_arg } :: l ->
+  let with_protect f =
+    let defs = ref [] in
+    let protect name (lam, layout) =
+      match lam with
+        Lvar _ | Lconst _ -> (lam, layout)
+      | _ ->
+          let id = Ident.create_local name in
+          defs := (id, layout, lam) :: !defs;
+          (Lvar id, layout)
+    in
+    let body = f protect in
+    List.fold_right
+      (fun (id, layout, lam) body -> Llet(Strict, layout, id, lam, body))
+      !defs body
+  in
+  let lfunction_helper ~mode_closure ~mode_arg ~mode_ret loc
```

Comment:
I was about to suggest to call this `build_unary_function_for_elided_argument` but then that made me realize... this only builds unary functions! That's not good. We need `f _ _` to be a two-argument function for efficiency.

Reply:
Yes that makes sense. I will rewrite this part into mutli-arguments function.